src/rust/e2e/src/utils.rs

```diff
@@ -0,0 +1,133 @@
+use std::{time::Duration, str::FromStr};
+
+use anyhow::Result;
+use aptos_sdk::{
+    bcs,
+    move_types::{
+        ident_str,
+        language_storage::{ModuleId, StructTag, TypeTag},
+    },
+    rest_client::{aptos_api_types::MoveModuleId, FaucetClient},
+    types::{
+        account_address::AccountAddress, transaction::EntryFunction, LocalAccount
+    },
+};
+use clap::Parser;
+use econia_sdk::EconiaClient;
+use sqlx::PgPool;
+
+pub const TIMEOUT: Duration = Duration::from_secs(1);
+
+#[derive(Parser, Debug)]
+pub struct Args {
+    /// The URL of the Aptos node
+    pub node_url: String,
+
+    /// The URL of the faucet
+    pub faucet_url: String,
+
+    /// The address of the Econia contract
+    pub econia_address: String,
+
+    /// The address of the Aptos faucet
+    pub faucet_address: String,
+
+    /// The database URL
+    pub db_url: String,
+}
+
+pub struct State {
+    pub e_apt: TypeTag,
+    pub e_usdc: TypeTag,
+    pub faucet_address: AccountAddress,
+    pub faucet_client: FaucetClient,
+    pub econia_address: AccountAddress,
+    pub econia_client: EconiaClient,
+    pub db_pool: PgPool,
+    pub node_url: String,
+}
+
+/// Creates the initial variables needed
+pub async fn init(args: &Args) -> State {
+    // Create eAPT and eUSDC `TypeTag`s
+    let faucet_address = AccountAddress::from_hex_literal(&args.faucet_address).unwrap();
+    let e_apt = TypeTag::Struct(Box::new(
+        StructTag::from_str(&format!("0x{faucet_address}::example_apt::ExampleAPT")).unwrap(),
+    ));
+    let e_usdc = TypeTag::Struct(Box::new(
+        StructTag::from_str(&format!("0x{faucet_address}::example_usdc::ExampleUSDC")).unwrap(),
+    ));
+
+    // Create a `FaucetClient`
+    let faucet_client = FaucetClient::new(
+        reqwest::Url::parse(&args.faucet_url).unwrap(),
+        reqwest::Url::parse(&args.node_url).unwrap(),
+    );
+
+    // Transform the Econia address from `String` to `AccountAddress`
+    let econia_address =
+        AccountAddress::from_hex_literal(&args.econia_address).expect("Could not parse address.");
+
+    let (_, econia_client) = account(&faucet_client, &args.node_url, econia_address.clone()).await;
+
+    let db_pool = PgPool::connect(&args.db_url)
+        .await
+        .expect("Could not connect to the database.");
+
+    State {
```

Comment:
State will probably need a thread-safe integer for the min size of the market, which gets accessed and increased with every test. This way each test can create its own market safely.